src/payment/strategies/humm/humm-payment-strategy.ts

```diff
@@ -0,0 +1,32 @@
+import { InternalCheckoutSelectors } from '../../../checkout';
+import { OrderRequestBody } from '../../../order';
+import { PaymentArgumentInvalidError } from '../../errors';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import { ExternalPaymentStrategy } from '../external';
+
+export default class HummPaymentStrategy extends ExternalPaymentStrategy {
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment?.methodId) {
+            throw new PaymentArgumentInvalidError(['payment.methodId']);
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+        try {
+            return await this._store.dispatch(this._paymentActionCreator.submitPayment({methodId: payment.methodId}));
+        } catch (error) {
+            if (!this._isAdditionalActionRequired(error)) {
+                return Promise.reject(error);
+            }
+
+            if (!error.body.provider_data) {
+                throw new PaymentArgumentInvalidError(['payment.provider_data']);
+            }
+
+            return new Promise(() => {
+                this._formPoster.postForm(error.body.additional_action_required.data.redirect_url, JSON.parse(error.body.provider_data));
```

Comment:
why is `error.body.provider_data` a string that we need to parse as JSON

why not just have an actual JSON response?